src/api/data/tree/exportApi.ts

```diff
@@ -111,45 +112,32 @@ export class ExportApi {
    */
   private validate(config: ExportConfig): ExportConfig {
     if (!config.query) {
-      const e = messages.createError('queryNotProvided');
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore override readonly .name field
-      e.name = 'queryNotProvided';
-      throw e;
+      throw new SfError(messages.getMessage('queryNotProvided'), 'queryNotProvided');
     }
 
     const filepath = path.resolve(process.cwd(), config.query);
     if (fs.existsSync(filepath)) {
       config.query = fs.readFileSync(filepath, 'utf8');
 
       if (!config.query) {
-        const e = messages.createError('queryNotProvided');
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-ignore override readonly .name field
-        e.name = 'queryNotProvided';
-        throw e;
+        throw messages.createError('queryNotProvided');
       }
     }
 
     config.query = config.query.toLowerCase().trim();
     if (!config.query.startsWith('select')) {
-      const e = messages.createError('soqlInvalid', [config.query]);
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore override readonly .name field
-      e.name = 'soqlInvalid';
-      throw e;
+      throw new SfError(messages.getMessage('soqlInvalid', [config.query]), 'soqlInvalid');
     }
 
     return config;
   }
 
   private setupOutputDirectory(outputDir: string): void {
     try {
-      fs.mkdirSync(outputDir);
+      mkdirpSync(outputDir);
```

Comment:
if you've already got fs-extra in the repo, then you could use
ensureDir and not have to try/catch the EEXIST stuff.

https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir-sync.md